agent/pool/pool.go

```diff
@@ -78,12 +119,14 @@ func (c *Conn) getClient() (*StreamClient, error) {
 		return nil, err
 	}
 
+	timeoutStream := &TimeoutConn{stream, c.pool.Timeout, 0, 0}
```

Comment:
It would be more readable to explicitly define struct fields here

Reply:
Do we ever set WriteTimeout anywhere else in this PR? Wondering if it's valuable to keep read/write timeouts separate (my naive assumption would be that the upper bound on size of reads and writes wouldn't vary enough to benefit from tuning them separately)

Reply:
`WriteTimeout` isn't used anywhere, so I can remove `WriteTimeout` and just make `TimeoutConn.Write` pass through to `c.Conn.Write`.

The read timeout needs to be kept separate from writes though, because the RPC timeout is read-specific, it applies to the first stream `Read` following the RPC request.